objectives-and-ambition.tex

```diff
@@ -0,0 +1,337 @@
+  automatically} by executing some kind of script or program that is part of the
+archive. First, if the automatic execution is possible, we know that there is a
+complete description of the protocol included in the archive, and that no
+mistake is made in trying to follow the protocol. Second, the automatic
+execution saves time.
+
+\emph{Given enough training or experience}, researchers can use existing technical
+solutions to compose such archives, and to allow a completely automatic
+reproduction of the necessary software environment, and then reproduction of the actual
+result using that software. 
+
+\medskip Researchers who use the Jupyter Notebook to orchestrate their
+computational research (see Figure~\ref{fig:jeodpp}) can achieve this automatic
+reproducibility with little additional effort: they use the Notebook document as
+the protocol of their analysis (Step 2), which can be executed automatically.
+They can make use of the Binder tool (Section~\ref{seq:project-binder}) and the
+associated mybinder.org service that has
+been designed by the Jupyter team to automatically create the appropriate
+software environment (step 1) in which the notebook can be executed.
+
+\begin{figure}[tb]
+  \centering\includegraphics[height=0.2\textheight]{images/jeodpp.png}
+  \centering\includegraphics[height=0.2\textheight]{images/jeodpp-demo.jpg}
+  \caption{\emph{Left}: The Joint Research Centre (JRC) Earth Observation
+    Data and Processing Platform (JEODPP) is a user of the
+    Jupyter Notebook (source:
+    \url{https://cidportal.jrc.ec.europa.eu/home/}), where it features
+    at the top of the pyramid to help users orchestrate layers of data
+    analysis software and hardware. \emph{Right}: An example
+    service in which an interactive visualisation is provided through
+    the Jupyter Notebook rendering of the density map of the ships
+    detected from Sentinel-1 images over the Mediterranean sea during
+    the period October 2014 to September 2016. \cite[Figure
+    6]{Soille2018}. \label{fig:jeodpp}}
+  \TODO{Do we want to keep the figure? The left side is useful, and it is good
+    to have some images.}
+\end{figure}
+
+\subsubsection{Beyond state of the art}
+
+In this project, we will focus on the \emph{reproduction of the
+  software environment} (Step 1) which is a prerequisite for any attempt to
+reproduce the actual research outputs. In particular, we want to make the
+creation of this computational environment \emph{automatic}.
+
+We will go beyond the current state of the art by
+\begin{compactitem}
+\item enhancing the existing Binder software for Jupyter notebooks,
+  directly improving the reproducibility of research created by the substantial
+  community of researchers using such Notebooks,
+\item extending the capabilities of the Binder tools so that its capability to
+  create arbitrary software environments automatically can be used outside the
+  Jupyter ecosystem,
```

Comment:
I think we should tweak this a little bit, because it suggests it only works for notebooks today and we will fix that, when it's not true. It's more of a community issue - we've seen good success with notebook users, but already produce a tool that can be used beyond notebooks. We will _improve_ that experience with demonstrations and examples.

```suggestion
\item enhancing the existing Binder software, already widely used by authors of Jupyter notebooks,
  directly improving the reproducibility of research created by the substantial
  community of researchers using such notebooks,
\item extending the capabilities of the Binder tools so that its capability to
  create arbitrary software environments automatically can be realised more broadly beyond the
  Jupyter user community,
```